### PR TITLE
docs: Authoring devfile version 1 and 2

### PR DESCRIPTION
**Pull request by yhontyk**
Introducing "Authoring devfile version  2":

* Distinguish Authoring devfile for version 1 and 2.
* Documenting both versions.
* Handling redirections from previous URLs.

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [x] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [x] Changed article references are updated where they are used (or a redirect has been configured on the docs side):
    - [x] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/src/services/bootstrap/branding.constant.ts)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

File: modules/end-user-guide/pages/authoring-devfiles-version-1.adoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[id="authoring-devfiles-version-1"]
+// 
+:navtitle: Authoring devfiles version 1
+:keywords: end-user-guide, authoring-devfiles-verison-1
```
Comment (themr0c) [inline modules/end-user-guide/pages/authoring-devfiles-version-1.adoc]:
```suggestion
:keywords: end-user-guide, authoring-devfiles-verison-1
```
```suggestion
:keywords: end-user-guide, authoring-devfiles-version-1
```

File: modules/end-user-guide/pages/authoring-devfiles-version-1.adoc [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+:page-aliases: .:authoring-devfiles-verison-1
```
Comment (themr0c) [inline modules/end-user-guide/pages/authoring-devfiles-version-1.adoc]:
```suggestion
:page-aliases: .:authoring-devfiles-version-1
```

File: modules/end-user-guide/pages/authoring-devfiles-version-2.adoc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[id="authoring-devfiles-version-2"]
+// 
+:navtitle: Authoring devfiles version 2
+:keywords: end-user-guide, authoring-devfiles-verison-2
```
Comment (themr0c) [inline modules/end-user-guide/pages/authoring-devfiles-version-2.adoc]:
```suggestion
:keywords: end-user-guide, authoring-devfiles-version-2
```

**Comment by themr0c** [comment]
Waiting for broken link fix in  https://github.com/eclipse/che-docs/pull/2003

**Review by themr0c — APPROVED** [review]

**Review by ericwill — APPROVED** [review]

**Review by themr0c — APPROVED** [review]

**Comment by themr0c** [comment]
Reopened there after erroneous repository deletion:  https://github.com/eclipse/che-docs/pull/2012

**Pull request closed without merging**